lib/taurus/qt/qtgui/taurusgui/config_loader/__init__.py

```diff
@@ -0,0 +1,63 @@
+#############################################################################
+##
+# This file is part of Taurus
+##
+# http://taurus-scada.org
+##
+# Copyright 2011 CELLS / ALBA Synchrotron, Bellaterra, Spain
+##
+# Taurus is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+##
+# Taurus is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+##
+# You should have received a copy of the GNU Lesser General Public License
+# along with Taurus.  If not, see <http://www.gnu.org/licenses/>.
+##
+###########################################################################
+
+"""
+This module provides factory 'getLoader' for proper ConfigLoader object
+detected from 'confname' string. Each config loader has to implement interface
+defined by AbstractConfigLoader class in
+taurus.qt.qtgui.taurusgui.config_loader.abstract
+"""
+
+import pkg_resources
+
+from taurus import warning
+
+__all__ = ["getLoader"]
+
+
+def getLoader(confname):
+    """
+    Discover proper config loader based on passed string.
+    It can be either path to file or directory or Python
+    abolute path to module with configuration.
+
+    :param confname: name of configuration
+    :return: A AbstractConfigLoader subclass object
+    """
+
+    loaders = []
+    for ep in pkg_resources.iter_entry_points("taurus.gui.loaders"):
```

Comment:
Tip: to facilitate testing with entry points, I found that a good trick is to create a variable with the entry-point group name (e.g. doing):

```python
EP_GROUP_LOADERS = "taurus.gui.loaders"
for ep in pkg_resources.iter_entry_points(EP_GROUP_LOADERS):
    ...
```

In this way the tests can use the `monkeypatch`pytest  fixture to change EP_GROUP_LOADERS to some temporary value in order to have a more controlled and isolated test environment

Reply:
Good point, will do.